code/game/objects/items/weapons/swords_axes_etc.dm

```diff
@@ -131,25 +131,13 @@
 		w_class = initial(w_class)
 		force = initial(force) //not so robust now
 		attack_verb = list("hits", "punches")
+	for(var/I in src.overlays)
+		var/image/P = I
+		if(P.name == "blood_overlay")
+			set_blood_overlay()
```

Comment:
There really wasn't a better way to do this besides adding a variable to atoms that indicate whether or not they're currently meant to be bloodied, although I'll rework this if someone has a suggestion.

Reply:
You could keep `generate_blood_overlay()` as is for items generally, and fix things for just for the telebaton. The main reason for the bug is that the baton's `attack_self()` always adds a blood overlay, even if there's no blood on the baton.
https://github.com/vgstation-coders/vgstation13/blob/9bbc953ff111d2fa0e9929396fbe26be78fbb82f/code/game/objects/items/weapons/swords_axes_etc.dm#L137-L143
Something like putting the above overlay generation behind a check for `if(blood_overlay)` could probably work.

Reply:
I tried this, but it didn't work because `blood_overlay` is not cleared when the blood overlay in the overlay list is removed. Now, it'd be trivial to simply make this blood overlay go away when the blood is cleaned, but the intention behind this blood overlay code is clearly that blood overlays are cached, and that we also don't want to keep making / retrieving new ones unnecessarily. I don't know if it's because `blend()` is expensive or something, but I figured that was fair enough.

As for the `generate_blood_overlay()` code, I mostly kept it the same, although I did improve it by simply having an item's blood overlay update and skip the rest of the code, if that's what was needed. I also changed the global `blood_overlays` list to have things stored as `"[type][icon_state]"` instead of just `"[type]"`, which will make solving this issue much easier in case another item like the telebaton is ever added, which needs a separate blood overlay image for each icon state.

Although this was definitely a convoluted way to solve a simple issue for the telebaton, the hope is that these changes are more future-proof than the current system.